### PR TITLE
Repair support for RocksDB

### PR DESCRIPTION
**Pull request by je4d**
Further integrate the RocksDB option into cmake, and use the ldb abstratcion in parts of the code that were using leveldb directly

### CODE REVIEW

**Comment by chfast** [comment]
Hi @je4d,
Can you tell us why are you working on this?

**Comment by je4d** [comment]
I was trying to sync the blockchain, and at some point it slowed to a crawl. It seemed to be spending most of the time reading from leveldb whilst executing contracts. I read something that indicated that rocksdb may be better suited to eth's workload, so I decided to give it a go, and I had to make changes to get it to build. That's about it.

**Comment by chfast** [comment]
Great. I created https://github.com/ethereum/cpp-ethereum/issues/4037 some time ago, so your work is very helpful.

File: libdevcore/CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,13 @@ target_include_directories(devcore SYSTEM PUBLIC ${BOOST_INCLUDE_DIR})
 add_dependencies(devcore BuildInfo.h)
 
 if (NOT EMSCRIPTEN)
-    find_package(LevelDB)
-    target_include_directories(devcore SYSTEM PUBLIC ${LEVELDB_INCLUDE_DIRS})
-    target_link_libraries(devcore Boost::Filesystem Boost::Random Boost::Thread ${LEVELDB_LIBRARIES})
+    if (ROCKSDB)
```
Comment (chfast) [inline libdevcore/CMakeLists.txt]: Can you separate common dependencies from conditional leveldb / rocksdb ones? Also `if (NOT EMSCRIPTEN)` can be remove as it is not needed any more.

File: libdevcore/dbfwd.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file dbfwd.h
+ * @author Jeff Snyder <jeff-github@caffeinated.me.uk>
```
Comment (chfast) [inline libdevcore/dbfwd.h]: We don't continue adding `@file` section if no additional description included, but it is fine if you want keep you name here. In any case remove `dbfwd.h` argument from `@file` tag and the `@date` tag.

**Pull request merged**